File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        result=True
        def height(node):
            nonlocal result
            if not node:
                return 0
        
            left=height(node.left)
            right=height(node.right)
            if abs(left-right)>1:
                result=False
            return max(left,right)+1
        height(root)
        return result
        
        
        
        height(root)

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    int depth(TreeNode temp)
    {
        if(temp==null)
        {
            return 0;
        }
        return 1+ Math.max(depth(temp.left),depth(temp.right));
    }

    public int maxDepth(TreeNode root) {
    return depth(root);
    }
}
